code/game/turfs/simulated/floor.dm

```diff
@@ -78,11 +78,43 @@ GLOBAL_LIST_INIT(icons_to_ignore_at_floor_init, list("damaged1","damaged2","dama
 		if(A.level == 3)
 			return 1
 
+/turf/simulated/proc/read_air()
+	if(blocks_air)
+		var/datum/gas_mixture/air = new()
+		return air
+	return read_milla_air()
+
+// From MILLA/src/lib.rs
+// Increased by 1 due to the difference in array indexing.
+#define GAS_OFFSET 7
+// Rust deals in thermal energy, but converts when talking to DM.
+#define ATMOS_TEMPERATURE 13
+#define ATMOS_INNATE_HEAT_CAPACITY 18
+/turf/proc/read_milla_air()
+	var/datum/gas_mixture/air = new()
+	var/list/raw_atmos = get_tile_atmos(x, y, z)
+	// Numbers from MILLA/src/lib.rs, plus one due to array indexing.
+	air.oxygen = raw_atmos[GAS_OFFSET + 0]
+	air.carbon_dioxide = raw_atmos[GAS_OFFSET + 1]
+	air.nitrogen = raw_atmos[GAS_OFFSET + 2]
+	air.toxins = raw_atmos[GAS_OFFSET + 3]
+	air.sleeping_agent = raw_atmos[GAS_OFFSET + 4]
+	air.agent_b = raw_atmos[GAS_OFFSET + 5]
+	air.temperature = raw_atmos[ATMOS_TEMPERATURE]
+	air.innate_heat_capacity = raw_atmos[ATMOS_INNATE_HEAT_CAPACITY]
+	return air
+#undef GAS_OFFSET
+#undef ATMOS_TEMPERATURE
+#undef ATMOS_INNATE_HEAT_CAPACITY
+
+/turf/simulated/proc/write_air(var/datum/gas_mixture/air)
```

Comment:
```suggestion
/turf/simulated/proc/write_air(datum/gas_mixture/air)
```
Var is actually implied here. Nice dm feature that isn't super well documented

Reply:
Yeah, but they tend to creep in because you're writing var/whatever everywhere else.